### PR TITLE
more interface changes, makin' this api a bagin'

### PR DESCRIPTION
**Pull request by ababkin**
depends on https://github.com/SumAll/haskell-eureka-client/pull/18

### CODE REVIEW

File: eureka-semver.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name:                eureka-semver
-version:             0.0.3.2
+version:             0.0.3.3
```
Comment (MHova) [inline eureka-semver.cabal]: This must also be a major version change.

File: src/Network/Eureka/Version/Cabal.hs [hunk 2]
```diff
@@ -24,3 +25,16 @@ fromString =
...
+never _ = False
+
+gte :: String -> Predicate
+gte = maybe never atLeastVersion . fromString
+  where
+    atLeastVersion :: Version -> Predicate
+    atLeastVersion version =
+      maybe
+        True -- if the version is Nothing
+        (>= version)
```
Comment (MHova) [inline src/Network/Eureka/Version/Cabal.hs]: Does this need to be `<` instead? `>= x y` returns true if `x` is greater than or equal to `y`, right? Here, `x` is required `version` and `y` is the evaluated version, yes? If so, do we need `(< version)` here?

Comment (MHova) [inline src/Network/Eureka/Version/Cabal.hs]: Oh wait is the `Maybe Version` in the `Predicate` the required version or the version-to-be-evaluated?

**Comment by ababkin** [comment]
@MHova you're right sir, i'll miss your vigilance. I don't think your next cushy corporate environment will appreciate this as much. :-)
BTW, just spoke with Ethan, and will be rewriting this using template haskell

**Pull request merged**